<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>839</width>
    <height>723</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBoxVisible">
        <property name="title">
         <string>Visible</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QCheckBox" name="checkBoxPieces">
           <property name="cursor">
            <cursorShape>ArrowCursor</cursorShape>
           </property>
           <property name="text">
            <string>Pieces</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkBoxChessboard">
           <property name="cursor">
            <cursorShape>ArrowCursor</cursorShape>
           </property>
           <property name="text">
            <string>Chessboard</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBoxLightVectors">
        <property name="title">
         <string>Light Vectors</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QCheckBox" name="checkBoxVectorN">
           <property name="text">
            <string>Vector n</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkBoxVectorL">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Vector l</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkBoxVectorV">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Vector v</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="checkBoxVectorR">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Vector r</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBoxView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>View</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QRadioButton" name="radioButtonPlayer1">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="cursor">
            <cursorShape>ArrowCursor</cursorShape>
           </property>
           <property name="text">
            <string>Player 1</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButtonPlayer2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="cursor">
            <cursorShape>ArrowCursor</cursorShape>
           </property>
           <property name="text">
            <string>Player 2</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButtonCrowd">
           <property name="cursor">
            <cursorShape>ArrowCursor</cursorShape>
           </property>
           <property name="text">
            <string>Crowd</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="radioButtonPawn">
           <property name="cursor">
            <cursorShape>ArrowCursor</cursorShape>
           </property>
           <property name="text">
            <string>Pawn</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBoxCamera">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Camera</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <property name="leftMargin">
          <number>9</number>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBoxEye">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>Eye</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="topMargin">
             <number>9</number>
            </property>
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBoxEyeX">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="cursor">
               <cursorShape>ArrowCursor</cursorShape>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
              <property name="value">
               <double>-8.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBoxEyeY">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBoxEyeZ">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
              <property name="value">
               <double>8.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBoxLookAt">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>Look At</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBoxLookAtX">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBoxLookAtY">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBoxLookAtZ">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBoxViewUp">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>View Up Vector</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBoxViewUpX">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
              <property name="value">
               <double>-8.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBoxViewUpY">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="doubleSpinBoxViewUpZ">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-10.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.250000000000000</double>
              </property>
              <property name="value">
               <double>9.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <widget class="PbGLWidget" name="widgetGLWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QSlider" name="horizontalSliderNormalSize">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="cursor">
       <cursorShape>OpenHandCursor</cursorShape>
      </property>
      <property name="mouseTracking">
       <bool>false</bool>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>500</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>839</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PbGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">PbGLWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>radioButtonCrowd</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateView()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>791</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButtonPawn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateView()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>418</y>
    </hint>
    <hint type="destinationlabel">
     <x>761</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButtonPlayer2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateView()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>362</y>
    </hint>
    <hint type="destinationlabel">
     <x>803</x>
     <y>202</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButtonPlayer1</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateView()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>771</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxEyeX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>690</x>
     <y>516</y>
    </hint>
    <hint type="destinationlabel">
     <x>753</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxEyeY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>516</y>
    </hint>
    <hint type="destinationlabel">
     <x>741</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxEyeZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>800</x>
     <y>516</y>
    </hint>
    <hint type="destinationlabel">
     <x>752</x>
     <y>356</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxLookAtZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>800</x>
     <y>583</y>
    </hint>
    <hint type="destinationlabel">
     <x>740</x>
     <y>422</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxLookAtY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>583</y>
    </hint>
    <hint type="destinationlabel">
     <x>743</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxLookAtX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>690</x>
     <y>583</y>
    </hint>
    <hint type="destinationlabel">
     <x>740</x>
     <y>395</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxViewUpX</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>690</x>
     <y>650</y>
    </hint>
    <hint type="destinationlabel">
     <x>744</x>
     <y>475</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxViewUpY</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>745</x>
     <y>650</y>
    </hint>
    <hint type="destinationlabel">
     <x>739</x>
     <y>470</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>doubleSpinBoxViewUpZ</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>800</x>
     <y>650</y>
    </hint>
    <hint type="destinationlabel">
     <x>828</x>
     <y>484</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxChessboard</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateVisible()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>828</x>
     <y>112</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxPieces</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateVisible()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>828</x>
     <y>88</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSliderNormalSize</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>normalSize(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>697</y>
    </hint>
    <hint type="destinationlabel">
     <x>231</x>
     <y>602</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxVectorN</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateLightVectors()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>828</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxVectorL</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateLightVectors()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>203</y>
    </hint>
    <hint type="destinationlabel">
     <x>828</x>
     <y>201</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxVectorV</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateLightVectors()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>828</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxVectorR</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>updateLightVectors()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>809</x>
     <y>262</y>
    </hint>
    <hint type="destinationlabel">
     <x>828</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateCamera()</slot>
  <slot>updateView()</slot>
  <slot>updateVisible()</slot>
  <slot>rotateCamera()</slot>
  <slot>rotateCamera(int)</slot>
  <slot>normalSize(int)</slot>
  <slot>updateLightVectors()</slot>
 </slots>
</ui>
